esp/platform/txsummary.cpp

```diff
@@ -663,3 +684,67 @@ CTxSummary::TxEntryBase* CTxSummary::queryEntry(const char* key)
 
     return nullptr;
 }
+
+StringBuffer& CTxOpenTelemetryConnector::normalizeKey(const char* txKey, const char* otKey, StringBuffer& normalized) const
+{
+    getSnakeCase(normalized, isEmptyString(otKey) ? txKey : otKey);
+    // Keys for custom attributes may be annotated to reflect the source here.
+    return normalized;
+}
+
+bool CTxOpenTelemetryConnector::isExcluded(const char* key, LogLevel logLevel, unsigned groupMask) const
+{
+    if ((logLevel > maxLogLevel) || ((groupMask & groupSelector) != groupSelector) || isEmptyString(key))
+        return true;
+    // Redundant attributes may be excluded by key here. For example, if spans include values like
+    // the global ID, it may be preferred to not forward a trace summary-sourced copy.
+    return false;
+}
+
+#define ENSURE_TARGET_AND_NAME \
```

Comment:
what's the advantage of encapsulating this logic in a define?

Reply:
Five identical blocks of setup logic with incredibly simple usage. It could be put into a method returning target (or nullptr) if that's preferred.